include/nbl/ext/DebugDraw/CDraw3DLine.h

```diff
@@ -54,26 +53,55 @@ class CDraw3DLine : public core::IReferenceCounted, public core::InterfaceUnmova
 		)
 		{
 			m_lines = core::vector<std::pair<S3DLineVertex, S3DLineVertex>>{ std::pair(S3DLineVertex{{ fromX, fromY, fromZ }, { r, g, b, a }}, S3DLineVertex{{ toX, toY, toZ }, { r, g, b, a }}) };
-			recordToCommandBuffer();
+			updateMeshBuffer();
```

Comment:
don't update gpu buffer with every line data modification, have another public method like commitLineData()

Reply:
actually just have this updateMeshBuffer public (update whole meshbuffer, not just vertex data) like you have now